types/jest/index.d.ts

```diff
@@ -1463,11 +1463,15 @@ declare namespace jest {
 
     type MockResult<T> = MockResultReturn<T> | MockResultThrow | MockResultIncomplete;
 
-    interface MockContext<T, Y extends any[]> {
+    interface MockContext<T, Y extends any[], C = any> {
```

Comment:
Why generic type argument is needed?

Reply:
Because, while using `Jest.fn`, the type of `this` can be specified, as it is in the two examples you shown.

In case it is specified by the user, I think it is a ggod idea to make `conexts` property to have the same type.

Reply:
Sounds good. Could you add a test from user perspective to check that all works as expected? Simply turn this explanation into code.

By the way, because of historical reasons Jest built-in types default to `unknown`, but in `@types/jest` the default is `any`.

Reply:
Nice suggestion @mrazauskas !
This test made the evidence af a lot of still missing stuff.
4e15331f7a should solve this.